internal/controllers/clusterextension_controller.go

```diff
@@ -193,15 +193,16 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	l.V(1).Info("handling finalizers")
 	finalizeResult, err := r.Finalizers.Finalize(ctx, ext)
 	if err != nil {
-		// TODO: For now, this error handling follows the pattern of other error handling.
-		//  Namely: zero just about everything out, throw our hands up, and return an error.
-		//  This is not ideal, and we should consider a more nuanced approach that resolves
-		//  as much status as possible before returning, or at least keeps previous state if
-		//  it is properly labeled with its observed generation.
-		setInstallStatus(ext, nil)
 		setResolutionStatus(ext, nil)
 		setStatusProgressing(ext, err)
-		ensureAllConditionsWithReason(ext, ocv1alpha1.ReasonFailed, err.Error())
+
+		// Here we check if the bundle is already installed.
+		// We only set Installed status to False if the bundle is actually no longer on cluster.
+		installedBundle, _ := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
+		if installedBundle == nil {
+			setInstallStatus(ext, nil)
+			setInstalledStatusConditionFailed(ext, err.Error())
+		}
```

Comment:
Circling back around to this after spending some time doing some more thinking on this. I can't think of a scenario where we would want to actually modify any of the information in the status related to installation on finalizer issues, so I think we should just remove all logic related to install status here:
```suggestion

```

My thinking for context:
- If we have successfully installed the ClusterExtension before, we do nothing
- If we have not successfully installed the ClusterExtension before:
    - We should preserve installation failure messages if they are set
    - If we failed before ever setting the installed condition and this ClusterExtension is being deleted, there really isn't any point to populate the empty install status field+condition.

Reply:
Looks like this change didn't actually make it into the PR? I still see the logic that I think we could remove.

Reply:
weird, maybe i accidentally undid it before I pushed. thanks for spotting it.